docs/user-guide/concepts/agents/session-management.md

```diff
@@ -0,0 +1,265 @@
+agent("Hello, I'm a new user!")
+```
+
+### S3SessionManager
+
+For cloud-based persistence, especially in distributed environments, use the `S3SessionManager`:
+
+```python
+from strands import Agent
+from strands.session.s3_session_manager import S3SessionManager
+import boto3
+
+# Optional: Create a custom boto3 session
+boto_session = boto3.Session(region_name="us-west-2")
+
+# Create a session manager that stores data in S3
+session_manager = S3SessionManager(
+    session_id="user-456",
+    bucket="my-agent-sessions",
+    prefix="production/",  # Optional key prefix
+    boto_session=boto_session,  # Optional boto3 session
+    region_name="us-west-2"  # Optional AWS region (if boto_session not provided)
+)
+
+# Create an agent with the session manager
+agent = Agent(session_manager=session_manager)
+
+# Use the agent normally - state and messages will be persisted to S3
+agent("Tell me about AWS S3")
+```
+
+#### Required S3 Permissions
+
+To use the `S3SessionManager`, your AWS credentials must have the following S3 permissions:
+
+- `s3:PutObject` - To create and update session data
+- `s3:GetObject` - To retrieve session data
+- `s3:DeleteObject` - To delete session data
+- `s3:ListBucket` - To list objects in the bucket
+
+Here's a sample IAM policy that grants these permissions for a specific bucket:
+
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "s3:PutObject",
+                "s3:GetObject",
+                "s3:DeleteObject"
+            ],
+            "Resource": "arn:aws:s3:::my-agent-sessions/*"
+        },
+        {
+            "Effect": "Allow",
+            "Action": "s3:ListBucket",
+            "Resource": "arn:aws:s3:::my-agent-sessions"
+        }
+    ]
+}
+```
+
+## How Session Management Works
+
+The session management system in Strands Agents works through a combination of events, repositories, and data models:
+
+### 1. Session Persistence Triggers
+
+Session persistence is automatically triggered by several key events in the agent lifecycle:
+
+- **Agent Initialization**: When an agent is created with a session manager, it automatically restores any existing state and messages from the session.
+- **Message Addition**: When a new message is added to the conversation, it's automatically persisted to the session.
+- **Agent Invocation**: After each agent invocation, the agent state is synchronized with the session to capture any updates.
+- **Message Redaction**: When sensitive information needs to be redacted, the session manager can replace the original message with a redacted version while maintaining conversation flow.
+
+> **Warning**: Changes to agent state or messages are only persisted during these specific lifecycle events, not immediately when made directly. Direct modifications will only be saved after the next trigger event occurs.
```

Comment:
We should clarify that adding messages directly or updating them directly won't even be persisted would it?

Like if after `agent.invoke_async` if I manually add a new message, it will never be persisted, right?

Reply:
It's worth turning this into an mkdocs warning:

https://squidfunk.github.io/mkdocs-material/reference/admonitions/?h=warning#+type:warning